### ISSUE TITLE
Experimental Design [*Required label behavior on either/or components]

### ISSUE THREAD

**Issue by tygindraux** [department-of-veterans-affairs/vets-design-system-documentation#2819]
## What

Currently, in the [forms library pattern for identification info](https://staging.va.gov/mock-form-patterns/identification-information), the *Required label is on SSN. But if a user starts to type in the VA File number field, the required label moves.

We're proposing a change to the *Required label behavior for all either/or components, which includes but isn't limited to SSN/VA Field number.

This involves:
- moving the *Required label to the < legend > with the < h3 > "School state or residency" heading (for example)
- moving the "Please provide only one of the following" into the element

This pattern follows after other components that use the radio or checkbox component.

## Purpose

- We did [research in Jan 2024](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/ask-va/design/User%20research/01-2024%20Submit%20an%20inquiry/Findings.md#additional-insights), where we tested the proposed behavior. We found people consistently understood that **either** SSN or Service Number are required, not both:

> "One of the following. So if you did the social security, you wouldn't have to do the service number."

- Then, we did follow-on [research in April 2024](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/ask-va/design/User%20research/04-2024%20Progress%20bar/Findings.md#finding-4-school-information-fields-are-interpreted-inconsistently-without-more-details), where we tested the current Design System behavior. We found that people consistently were **confused**. They thought that school state is required and residency state isn't. The label "starts" on school state.

> "I guess the school state is required and residency state is not."

## Usage

- This could be used in place of the current [forms library pattern for identification info](https://staging.va.gov/mock-form-patterns/identification-information).
- It can apply to any field that requires either/or, like the school state or residency state in the Ask VA form we tested.
 
## Behavior

- A user can choose to provide either X field or Y field.
- The *Required label is "part of" the H3.
- If they choose to provide X, the *Required label does not move from the H3.

## Examples

- [Example in CodePen](https://codepen.io/Thomas-Michaud/pen/GRLbJem)
- [Example in Figma, include accessibility annotations](https://www.figma.com/design/aQ6JsjD4pvMxSVPAZHllMX/AVA-Page-Library?node-id=1-5100)

## Accessibility

- Listed under Examples.

## Guidance

- Consider this component when you have a question that requires "one of the following" between two field inputs.

## Research (optional)

- Listed under Purpose.

## Code (optional)

- Listed under Examples.

## Next steps

You may present your work to the Design System Council at an upcoming meeting. If you do not or cannot attend the Design Council Meeting, you can opt to get an asynchronous approval.

Submit requests to join an upcoming Design System Council meeting in #platform-design-system.  

During the meeting, the Design System Council Working Group will evaluate the request and make a decision.

If your request is approved, you can [add your component or pattern to the system](https://design.va.gov/about/contributing-to-the-design-system/add-a-component-or-pattern-once-approved). If you have any questions on how to add your component or pattern to the system, please reach out to the Design System Team at #platform-design-system.

**Comment by tygindraux** [comment]
FYI @t-michaud

**Comment by caw310** [comment]
This is on the agenda for the 5/30 DSC meeting.

**Comment by humancompanion-usds** [comment]
@jeana-adhoc and @rhasselle-oddball - We should have invited you to this DSC meeting. As I recall we had some trouble implementing the required label on the legend? Is that accurate? We agreed in this meeting that it would be best to move the required label to the legend for SSN vs. VA File number where only 1 is required to be filled out, rather than moving the label dynamically. This especially makes more sense for other instances of the same pattern where a user has to fill out one of two options, which Tyler and Thomas presented and are included as examples in the initial comment. Can one of you remind me if we tried having the required label in the legend and it didn't work or if there was a reason it couldn't go there? Thanks!

**Comment by humancompanion-usds** [comment]
> However we do have this pattern called use-forms-pattern, but even with that the error is on the field, not the top part

I'm not sure moving the error is required. I believe the desire is to just move the required label up. I don't know that it makes sense for an error to not be related to a specific field. Does "Forms pattern single" allow you to move the required label?

**Comment by rhasselle-oddball** [comment]
> > However we do have this pattern called use-forms-pattern, but even with that the error is on the field, not the top part
> 
> I'm not sure moving the error is required. I believe the desire is to just move the required label up. I don't know that it makes sense for an error to not be related to a specific field. Does "Forms pattern single" allow you to move the required label?

No, the required label can't be moved using "Forms pattern single". I added a ticket to investigate some possibilities on [vets-website/forms-system](https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/1373) side, but you have a point... if we do need to use forms single/multiple pattern, then we would also need an enhancement on DST side.
